PowerPointLabs/PowerPointLabs/Views/SpotlightSettingsDialogBox.xaml.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows.Forms;
+using System.Windows.Media;
+
+using PowerPointLabs.Utils;
+
+namespace PowerPointLabs.Views
+{
+    /// <summary>
+    /// Interaction logic for SpotlightSettingsDialogBox.xaml
+    /// </summary>
+    public partial class SpotlightSettingsDialogBox
+    {
+        public delegate void UpdateSettingsDelegate(float spotlightTransparency, float softEdge, System.Drawing.Color newColor);
+        public UpdateSettingsDelegate SettingsHandler { get; set; }
+
+        private Dictionary<String, float> softEdgesMapping = new Dictionary<string, float>
+        {
+            {"None", 0},
+            {"1 Point", 1},
+            {"2.5 Points", 2.5f},
+            {"5 Points", 5},
+            {"10 Points", 10},
+            {"25 Points", 25},
+            {"50 Points", 50}
+        };
+        private float lastTransparency;
+
+        public SpotlightSettingsDialogBox()
+        {
+            InitializeComponent();
+        }
+
+        public SpotlightSettingsDialogBox(float defaultTransparency, float defaultSoftEdge, System.Drawing.Color defaultColor)
+            : this()
+        {
+            lastTransparency = defaultTransparency;
+            spotlightTransparencyInput.Text = defaultTransparency.ToString("P0");
+            spotlightTransparencyInput.ToolTip = "The transparency level of the spotlight effect to be created.";
+
+            String[] keys = softEdgesMapping.Keys.ToArray();
+            float[] values = softEdgesMapping.Values.ToArray();
```

Comment:
We have to do the mapping every time the dialog box is opened. Is it possible to make them const Arrays?